filebeat/input/filestream/input.go

```diff
@@ -359,16 +368,24 @@ func (inp *filestream) eventFromMessage(m reader.Message, path string) beat.Even
 		},
 	}
 	fields.DeepUpdate(m.Fields)
+	m.Fields = fields
+
+	for _, proc := range inp.msgPostProc {
+		proc.PostProcess(&m)
+	}
 
 	if len(m.Content) > 0 {
-		if fields == nil {
-			fields = common.MapStr{}
+		if m.Fields == nil {
+			m.Fields = common.MapStr{}
+		}
+		if _, ok := m.Fields["message"]; !ok {
+			m.Fields["message"] = string(m.Content)
 		}
-		fields["message"] = string(m.Content)
 	}
```

Comment:
For the complete function body up until here I wonder, if this could be a separate parser we just append to the list of parsers? In that case the `beat.Event` could be extracted from the message directly.